### PR TITLE
Add the ACP component

### PR DESCRIPTION
**Pull request by iMattPro**
Depends on #3

### CODE REVIEW

**Comment by rxu** [comment]
Tests fail.

**Comment by iMattPro** [comment]
Because it depends on 3

**Comment by rxu** [comment]
Works fine with #3 altogether.
The one thing I wonder is should forum prefixes be available in subforums or not. That would be wise to avoid prefixes duplicating through the forums tree.
Maybe maake it an option?

**Comment by iMattPro** [comment]
I'm not sure I understand. Prefixes are available in sub forums. They're available in any forum... Just not categories or links.

**Comment by rxu** [comment]
What I meant is: if you create prefixes for the parent forum, they caoukd be used in its subforums.

**Comment by iMattPro** [comment]
@michaelcullum

File: controller/admin_controller.php [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+	/** @var int */
+	protected $forum_id;
+
+	/** @var string */
+	protected $u_action;
+
+	/**
+	 * Constructor
+	 *
+	 * @param manager  $manager
```
Comment (michaelcullum) [inline controller/admin_controller.php]: Lets be explicit about what these are please

File: controller/admin_controller_interface.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\topicprefixes\controller;
+
+/**
+ * Interface admin_controller_interface
+ */
+interface admin_controller_interface
```
Comment (michaelcullum) [inline controller/admin_controller_interface.php]: We really don't need interfaces for controllers...

**Comment by iMattPro** [comment]
Updated...And EPV is broken...

**Comment by michaelcullum** [comment]
I meant specifying the namespace?

**Comment by iMattPro** [comment]
That's what the Use statements are for ;)
I hate the verbosity of fully qualifying them

**Comment by michaelcullum** [comment]
I was only suggesting to do so in the docblock. Otherwise things like `manager` are rather ambiguous.

**Comment by iMattPro** [comment]
Updated

**Comment by iMattPro** [comment]
Right but that's why there's a description next to it ;)

**Comment by michaelcullum** [comment]
Why use a description when you could just use the FQCN and then it's even clearer? Pwease. :P

**Comment by iMattPro** [comment]
This is done this way throughout the entire extension already. (It's even been done in the core in some places where people use phpStorm, because when you import a class, phpStorm does it like this automatically)

**Comment by iMattPro** [comment]
One example in core: https://github.com/phpbb/phpbb/blob/master/phpBB/phpbb/attachment/delete.php#L68-L77

**Comment by michaelcullum** [comment]
Hmpf

**Pull request merged**